system-int-test/test/cmr/system_int_test/ingest/subscription_ingest_test.clj

```diff
@@ -43,6 +44,74 @@
                            (map #(select-keys % [:concept-id :extra-fields :metadata])))]
     (#'jobsub/process-subscriptions (system/context) subscriptions)))
 
+(deftest subscription-count-limit-ignores-old-subscriptions-test
+  (side/eval-form `(jobsub/set-subscriptions-limit! 2))
+  (let [coll1 (data-core/ingest-umm-spec-collection
+               "PROV1"
+               (data-umm-c/collection
+                {:ShortName "coll-exceed-limit-1"
+                 :EntryTitle "entry-title-exceed-limit-1"})
+               {:token "mock-echo-system-token"})
+        user1-token (echo-util/login (system/context) "user1")]
+     (testing "ingest on PROV1 with subscription request for subscriber-id below subscription limit" ;zzz
+       (let [concept (subscription-util/make-subscription-concept {:Query "platform=NOAA-5"
+                                                                   :Name "sub1"
+                                                                   :CollectionConceptId (:concept-id coll1)
+                                                                   :SubscriberId "user1"})
+             response (ingest/ingest-concept concept {:token user1-token})]
+         (is (= 201 (:status response)))
+         (is (mdb/concept-exists-in-mdb? (:concept-id response) (:revision-id response)))))
+     (testing "ingest on PROV1 with subscription change for subscriber-id below subscription limit"
+       (let [concept (subscription-util/make-subscription-concept {:Query "platform=NOAA-6"
+                                                                   :Name "sub1"
+                                                                   :CollectionConceptId (:concept-id coll1)
+                                                                   :SubscriberId "user1"})
+             response (ingest/ingest-concept concept {:token user1-token})]
+        (is (= 200 (:status response)))
+        (is (mdb/concept-exists-in-mdb? (:concept-id response) (:revision-id response)))))
+     (testing "ingest on PROV1 with subscription request for subscriber-id below subscription limit"
+       (let [concept (subscription-util/make-subscription-concept {:Query "platform=NOAA-7"
+                                                                   :Name "sub2"
+                                                                   :CollectionConceptId (:concept-id coll1)
+                                                                   :SubscriberId "user1"})
+             response (ingest/ingest-concept concept {:token user1-token})]
+        (is (= 201 (:status response)))
+        (is (mdb/concept-exists-in-mdb? (:concept-id response) (:revision-id response)))))))
```

Comment:
I think we only need 1 `testing` in here.  Something like.. `(testing "revisions are not included when calculating subscription count" ...)`

You could also move it `subscription-count-exceeds-limit-test` after the first test.  Create a revision of "sub1" and since you've already set the subscription limit to 1, it should succeed.   You can verify both revisions exist using `(is (mdb/concept-exists-in-mdb? (:concept-id response) 1))` and `(is (mdb/concept-exists-in-mdb? (:concept-id response) 2))` (I think revision-id starts from 1).

Reply:
Combined the tests in to the single `subscription-count-exceeds-limit-test`. I went with your suggestion and tested a revision of "sub1" after hitting the limit of 1 after the first ingest.